Plugins/AWSLambdaPackager/Plugin.swift

```diff
@@ -0,0 +1,344 @@
+        stderrPipe.fileHandleForReading.readabilityHandler = outputHandler
+
+        let process = Process()
+        process.standardOutput = stdoutPipe
+        process.standardError = stderrPipe
+        process.executableURL = URL(fileURLWithPath: executable.string)
+        process.arguments = arguments
+        if let workingDirectory = customWorkingDirectory {
+            process.currentDirectoryURL = URL(fileURLWithPath: workingDirectory.string)
+        }
+        process.terminationHandler = { _ in
+            // Read and pass on any remaining free-form text output from the plugin.
+            stderrPipe.fileHandleForReading.readabilityHandler?(stderrPipe.fileHandleForReading)
+            // Read and pass on any remaining messages from the plugin.
+            stdoutPipe.fileHandleForReading.readabilityHandler?(stdoutPipe.fileHandleForReading)
+        }
+
+        try process.run()
+        process.waitUntilExit()
+
+        // wait for output to be full processed
+        sync.wait()
+
+        if process.terminationStatus != 0 {
+            throw Errors.processFailed(process.terminationStatus)
+        }
+
+        return output
+    }
+}
+
+private struct Configuration {
+    public let outputDirectory: Path
+    public let products: [Product]
+    public let buildConfiguration: PackageManager.BuildConfiguration
+    public let staticallyLinkRuntime: Bool
+    public let verboseLogging: Bool
+    public let baseImage: String
+    public let version: String
+    public let applicationRoot: String
+
+    public init(context: PluginContext, arguments: [String]) {
+        self.outputDirectory = context.pluginWorkDirectory.appending(subpath: "\(AWSLambdaPackager.self)") // FIXME: read argument
+        self.products = context.package.products.filter { $0 is ExecutableProduct } // FIXME: read argument, filter is ugly
+        self.buildConfiguration = .release // FIXME: read argument
+        #if os(Linux)
+        self.staticallyLinkRuntime = true // FIXME: read argument
+        #else
+        self.staticallyLinkRuntime = false // FIXME: read argument, warn if set to true
+        #endif
+        self.verboseLogging = true // FIXME: read argument
+        let swiftVersion = "5.6" // FIXME: read dynamically current version
+        self.baseImage = "swift:\(swiftVersion)-amazonlinux2" // FIXME: read argument
+        self.version = "1.0.0" // FIXME: where can we get this from? argument?
+        self.applicationRoot = "/app" // FIXME: read argument
+    }
+}
+
+private enum Errors: Error {
+    case unsupportedPlatform(String)
+    case unknownProduct(String)
+    case unknownExecutable(String)
+    case buildError(String)
+    case failedWritingDockerfile
+    case processFailed(Int32)
+    case invalidProcessOutput
+}
+
+extension PackageManager.BuildResult {
+    // find the executable produced by the build
+    func executableArtifact(for product: Product) throws -> PackageManager.BuildResult.BuiltArtifact? {
```

Comment:
Should this be in PackagePlugin in the future?  Seems like a common thing to want.